Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStateDiscriminatorConvention.cs

```diff
@@ -33,39 +33,56 @@ class StreamProcessorStateDiscriminatorConvention : IDiscriminatorConvention
 
         /// <summary>
         /// Gets the correct type when deserializing objects from  <see cref="AbstractStreamProcessorState"/> collection
-        /// depending on the "Partitioned" field, which is set by this class.
+        /// depending on the given nominal type or "Partitioned" field.
         /// </summary>
         /// <param name="bsonReader">A <see cref="IBsonReader"/>.</param>
         /// <param name="nominalType">The nominal type.</param>
         /// <returns>The actual wanted type.</returns>
         public Type GetActualType(IBsonReader bsonReader, Type nominalType)
         {
-            var bookmark = bsonReader.GetBookmark();
-            bsonReader.ReadStartDocument();
-            var actualType = nominalType;
-            while (bsonReader.ReadBsonType() != BsonType.EndOfDocument)
+            if (nominalType == typeof(StreamProcessorState))
             {
-                var fieldName = bsonReader.ReadName();
-                if (fieldName == "Partitioned")
+                return typeof(StreamProcessorState);
```

Comment:
You already have the type in `nominalType`, so you could just return that...

Reply:
fixd